serviceregistration/consul/consul_service_registration.go

```diff
@@ -539,65 +437,52 @@ func (c *ConsulServiceRegistration) reconcileConsul(registeredServiceID string,
 	}
 
 	if err := agent.ServiceRegister(service); err != nil {
-		return "", errwrap.Wrapf(`service registration failed: {{err}}`, err)
+		return errwrap.Wrapf(`service registration failed: {{err}}`, err)
 	}
 
 	if err := agent.CheckRegister(sealedCheck); err != nil {
-		return serviceID, errwrap.Wrapf(`service check registration failed: {{err}}`, err)
+		return errwrap.Wrapf(`service check registration failed: {{err}}`, err)
 	}
 
-	return serviceID, nil
+	return nil
 }
 
 // runCheck immediately pushes a TTL check.
-func (c *ConsulServiceRegistration) runCheck(sealed bool) error {
+func (c *ServiceRegistration) runCheck() error {
+	c.stateLock.RLock()
+	defer c.stateLock.RUnlock()
+
 	// Run a TTL check
 	agent := c.Client.Agent()
-	if !sealed {
+	if !c.state.IsSealed {
 		return agent.PassTTL(c.checkID(), "Vault Unsealed")
 	} else {
 		return agent.FailTTL(c.checkID(), "Vault Sealed")
 	}
 }
 
-// fetchServiceTags returns all of the relevant tags for Consul.
-func (c *ConsulServiceRegistration) fetchServiceTags(active bool, perfStandby bool) []string {
-	activeTag := "standby"
-	if active {
-		activeTag = "active"
-	}
-
-	result := append(c.serviceTags, activeTag)
-
-	if perfStandby {
-		result = append(c.serviceTags, "performance-standby")
```

Comment:
Only should be on if it's a perf standby, not leader or non-performance cluster.